base_layer/core/src/consensus/consensus_manager.rs

```diff
@@ -202,11 +206,15 @@ impl ConsensusManagerBuilder {
     }
 
     /// Builds a consensus manager
-    pub fn build(mut self) -> ConsensusManager {
+    pub fn build(mut self) -> Result<ConsensusManager, String> {
```

Comment:
String does not implement Error so preferable not to use it as an error. I think create a thiserror type. You could use anyhow error in the GRPC at the application level if that is easier.